fluss-server/src/main/java/com/alibaba/fluss/server/metadata/TabletServerMetadataCache.java

```diff
@@ -99,6 +103,53 @@ public Optional<String> getPartitionName(long partitionId) {
         return serverMetadataSnapshot.getPartitionName(partitionId);
     }
 
+    public Optional<TableInfo> getTableInfo(long tableId) {
+        return serverMetadataSnapshot.getTableInfo(tableId);
+    }
+
+    public Optional<TableMetadata> getTableMetadata(TablePath tablePath) {
+        ServerMetadataSnapshot snapshot = serverMetadataSnapshot;
+        OptionalLong tableIdOpt = snapshot.getTableId(tablePath);
+        if (!tableIdOpt.isPresent()) {
+            return Optional.empty();
+        } else {
+            long tableId = tableIdOpt.getAsLong();
+            Optional<TableInfo> tableInfoOpt = snapshot.getTableInfo(tableId);
+            if (tableInfoOpt.isPresent()) {
+                TableInfo tableInfo = tableInfoOpt.get();
+                List<BucketMetadata> bucketMetadataList = new ArrayList<>();
+                snapshot.getBucketMetadata(tableId)
+                        .forEach(
+                                (bucketId, bucketMetadata) ->
+                                        bucketMetadataList.add(bucketMetadata));
```

Comment:
Why not get the list directly? `snapshot.getBucketMetadata(tableId).values()` or `new ArrayList<>(snapshot.getBucketMetadata(tableId).values())`.